lib/snowplow_tracker/emitters/cache.ex

```diff
@@ -0,0 +1,76 @@
+defmodule SnowplowTracker.Emitters.Cache do
+  @moduledoc """
+  Module which implements functions to interact with ETS
+  """
+
+  require Logger
+
+  alias :ets, as: Ets
+
+  @table Application.get_env(:snowplow_tracker, :table)
+  #@persist_every Application.get_env(:snowplow_tracker, :persist_every, 5000)
+  @path Application.get_env(:snowplow_tracker, :path, ".")
+  @lock "lock"
+
+  def init(table \\ @table) do
+    PersistentEts.new(
+      table,
+      "#{@path}/#{Atom.to_string(table)}.tab",
+      [:public, :set, :named_table, {:read_concurrency, true}]
+    )
+
+    Logger.log(:debug, "Table #{table} created successfully!")
+    {:ok, :success}
+  end
+
+  def insert(payload, table \\ @table) do
+    Ets.insert(table, payload)
+    {:ok, :success}
+  end
+
+  def match(table \\ @table) do
+    {
+      :ok,
+      Ets.match(table, {:"$1", :"$2", :"$3"})
+    }
+  end
+
+  def check_lock(table \\ @table) do
+    {
+      :ok,
+      Ets.lookup(table, @lock)
+    }
+  end
+
+  # coveralls-ignore-start
+  def release_lock(payload, table \\ @table)
+  # coveralls-ignore-stop
+
+  def release_lock({:ok, _msg}, table), do: delete_key(@lock, table)
+
+  def release_lock({:error, _}, _table), do: :ok
+
+  def set_lock(table \\ @table) do
+    try do
```

Comment:
this is not required, you can just use `rescue`!

Reply:
Not sure what you mean. `rescue` block without `try` block?